src/main/java/net/floodlightcontroller/routing/VirtualGatewayInstance.java

```diff
@@ -0,0 +1,303 @@
+        return interfaces.values().stream()
+                .filter(intf -> intf.getInterfaceName().equals(name))
+                .findAny();
+
+    }
+
+    public boolean isInterfaceBelongsGateway(String name) {
+        return interfaces.values().stream()
+                .anyMatch(inft -> inft.getInterfaceName().equals(name));
+    }
+
+    public boolean removeInterface(String interfaceName) {
+        if (!getInterface(interfaceName).isPresent()) {
+            return false;
+        }
+        else {
+            interfaces.remove(interfaceName);
+            return true;
+        }
+    }
+
+    public void clearInterfaces() {
+        this.interfaces.clear();
+    }
+
+    public void addSwitchMember(DatapathId dpid) { this.switchMembers.add(dpid); }
+
+    public void addNptMember(NodePortTuple npt) { this.nptMembers.add(npt); }
+
+    public void addSubnetMember(IPv4AddressWithMask subnet) { this.subsetMembers.add(subnet); }
+
+    public void removeSwitchMember(DatapathId dpid) {
+        if (!this.switchMembers.isEmpty()) {
+            for (DatapathId id : this.switchMembers) {
+                if (id.equals(dpid)) {
+                    this.switchMembers.remove(id);
+                }
+            }
+        }
+    }
+
+    public void removeNptMember(NodePortTuple npt) {
+        if (!this.nptMembers.isEmpty()) {
+            for (NodePortTuple nodePortTuple : this.nptMembers) {
+                if (nodePortTuple.equals(npt)) {
+                    this.nptMembers.remove(npt);
+                }
+            }
+        }
+    }
+
+    public void removeSubnetMember(IPv4AddressWithMask iPv4AddressWithMask) {
+        if (!this.subsetMembers.isEmpty()) {
+            for (IPv4AddressWithMask subnet : this.subsetMembers) {
+                if (subnet.equals(iPv4AddressWithMask)) {
+                    this.subsetMembers.remove(iPv4AddressWithMask);
+                }
+            }
+        }
+    }
+
+    public void removeSwitchFromInstance(DatapathId dpid) {
+        if (!this.switchMembers.isEmpty()) {
+            for (DatapathId id : this.switchMembers) {
+                if (id.equals(dpid)) {
+                    this.switchMembers.remove(id);
+                }
+            }
+        }
+
+        if (!this.nptMembers.isEmpty()) {
+            for (NodePortTuple npt : this.nptMembers) {
+                if (npt.getNodeId().equals(dpid)) {
+                    this.nptMembers.remove(npt);
+                }
+            }
+        }
+    }
+
+    public void clearSwitchMembers() {
+        if (!this.switchMembers.isEmpty()) {
+            this.switchMembers.clear();
+        }
+    }
+
+    public void clearNptMembers() {
+        if (!this.nptMembers.isEmpty()) {
+            this.nptMembers.clear();
+        }
+    }
+
+    public void clearSubnetMembers() {
+        if (!this.subsetMembers.isEmpty()) {
+            this.subsetMembers.clear();
+        }
+    }
+
+    public boolean isAGatewayInft(IPv4Address ip) {
```

Comment:
The abbreviation for interface is "intf"

Reply:
That's correct, thanks!